files/build_templates/docker_image_ctl.j2

```diff
@@ -108,6 +108,37 @@ function preStartAction()
     fi
 {%- elif docker_container_name == "snmp" %}
     $SONIC_DB_CLI STATE_DB HSET 'DEVICE_METADATA|localhost' chassis_serial_number $(decode-syseeprom -s)
+
+    # Wait for system ready. Continue after timeout if we won't hear anything
+    # from DB. Read timeout from config file and add two extra minutes on top of
+    # it.
+    CONFFILE=system_health_monitoring_config.json
+    PLATFORM=$(sonic-cfggen -d -v "DEVICE_METADATA['localhost']['platform']")
```

Comment:
> Try to use cache rather than using sonic-cfggen as sonic-cfggen is a costly operation.

> Please refer to https://github.com/sonic-net/sonic-buildimage/pull/17343

@dgsudharsan ok, will change that